lib/operations/command.rb

```diff
@@ -93,14 +93,22 @@
 #    on the context (either initial or the data loaded by the contract).
 #    Anything that has nothing to do with the user input validation
 #    should be implemented as a precondition.
-# 4. Operation itself implements the routine. It can create or update
+# 4. Idempotency check are running after preconditions and can return
+#    either Success() or Failure({}). In case of Failure, the operation
+#    body (and after calls) will be skept but the operatiuon result will
+#    be successful. Failure({}) can carry an additional context to make
+#    sure the opreation result context is going to be the same for case
+#    or normal operation execution and skipped operation body. The only
```

Comment:
I am not sure I get this sentence, it's a bit tangled.

`Failure({}) can carry an additional context to make`
`sure the operation result context is going to be the same for case or ` <- what case?
`or normal operation execution and skipped operation body.` <- after the missing 'case' this part reads a bit like some words were missing from it